ibattery.py

```diff
@@ -0,0 +1,237 @@
+from libqtile.widget import base
+from libqtile.log_utils import logger
+from libqtile.utils import send_notification
+from libqtile import bar
+
+import math
+import cairocffi as cairo
+import psutil
+
+
+class Battery(base._Widget):
+    """A widget to display a nice battery.
+
+    requirements: psutil
+    optional: dbus-next (used to send notification).
+    """
+
+    orientations = base.ORIENTATION_HORIZONTAL
+    defaults = [
+        (
+            "padding",
+            4,
+            "padding on either side of of the widget."
+        ),
+        (
+            "foreground",
+            "ffffff",
+            "Battery color in normal mode."
+        ),
+        (
+            "charging_fg",
+            "02a724",
+            "foreground color when battery is charging."
+        ),
+        (
+            "update_interval",
+            30,
+            "time to wait until the widgets refreshes."
+        ),
+        (
+            "low_foreground",
+            "ff0000",
+            "change color when battery is low."
+        ),
+        (
+            "warn_below",
+            10,
+            "battery level to indicate battery is low."
+        ),
+        (
+            "notify",
+            False,
+            "send a notification when battery is low."
+        ),
+        (
+            "notification_timeout",
+            10,
+            "time in seconds to display notification."
+        ),
+    ]
+
+    def __init__(self, **config):
+        self.widget_width = 36
+        base._Widget.__init__(self, bar.CALCULATED, **config)
+        self.add_defaults(Battery.defaults)
+
+        self.length = self.padding * 2 + self.widget_width
+        self.HEIGHT = 16  # widgets height
+        self.BAR_WIDTH = 30  # battery bar
```

Comment:
Just a suggestion but you may want to make the size of the icon configurable rather than having it hardcoded. Users may have different size bars etc.